.devcontainer/Dockerfile.devcontainer

```diff
@@ -0,0 +1,49 @@
+FROM node:16.18 AS ui-builder
+
+WORKDIR /mlflow
+COPY mlflow/server/js .
+RUN yarn install --silent && yarn build
```

Comment:
If we included JS modules, that would significantly increase the image size. To keep the image size compact, we just include pre-built MLflow UI assets, which allow us for `mlflow server` or `mlflow ui`. This is not ideal for UI development because we need to install JS dependencies every time we re-create a container.

Reply:
Is there a way to have an option for this or a specific image that's geared towards UI dev work? (maybe in a follow-up PR?)

Reply:
Yes, we can just add another `devcontainer.jon`, but another image means another maintenance cost. I'd consider adding it if there is a request from the community.